### ISSUE TITLE
under behavior incorrect

### ISSUE THREAD

**Issue by jberger** [marcusramberg/mojolicious-plugin-promiseactions#4]
When using this plugin, suddenly in an under used for validation, `return 0` counts the same as `return 1`. A work-around is to `return` with no arguments, however this very unexpected.

```
plugin "PromiseActions"; under sub { return 0 }; get "/" => { text => "ok" }; app->start' get / 
```

**Comment by jberger** [comment]
The problem is context propagation. In an under, the action is called in scalar context, so the fact that we return an array, you get the count of the number of return values. Therefore `return 0` in scalar context is `1` because you returned one value. Le sigh

**Comment by marcusramberg** [comment]
Does that mean we actually have to use wantarray after all? :(

**Comment by jberger** [comment]
Closed by #5
